dashboard/app/views/levels/_level_group.haml

```diff
@@ -6,64 +6,86 @@
-    - last_attempt = last_attempt_entry["result"] if last_attempt_entry
+-#
+  If we are looking at the level independent of a script, then let's just
+  show all pages of the LevelGroup at once.  Otherwise we'll show the actual
+  page.
+
+- if @script_level.nil?
+  - pages = (0..data["pages"].length-1).to_a
+- else
+  - current_page = params[:puzzle_page] ? (params[:puzzle_page].to_i - 1) : 0
+  - pages = [current_page]
+
+- pages.each do |page|
+
+  - unless data["title"].blank?
+    %h2= t('heading_x_of_y', heading: data["title"], x: page+1, y: data["pages"].length)
+    %br/
+
+  - page_levels = data["pages"][page]["levels"]
+
+  - if page == 0
+    - prior_page_count = 0
   - else
-    - last_attempt = ""
-  - level_class = level.class.to_s.underscore
-
-  - if level_class == "multi"
-    -# For students (or teachers doing PD), mark all answers incorrect, so we don't give anything away.
-    - unless current_user.try(:authorized_teacher?) && !@script.try(:pd?)
-      - level.properties['answers'].each { |answer| answer['correct'] = false }
-    = render partial: 'levels/single_multi', locals: {page: page, index: index, standalone: false, level: level, last_attempt: last_attempt, tight_layout: true }
-  - elsif level_class = "text_match"
-    -# For students (or teachers doing PD), clear out all answers.
-    - unless current_user.try(:authorized_teacher?) && !@script.try(:pd?)
-      - level.properties['answers'] = nil
-    = render partial: 'levels/single_text_match', locals: {page: page, index: index, standalone: false, level: level, last_attempt: last_attempt }
+    - prior_page_count = data["pages"][0..page-1].inject(0){|sum,p| sum + p["levels"].length}
 
-= render partial: 'levels/common_audio'
+  - page_levels.each_with_index do |level_name, index|
+    - level = Level.find_by_name(level_name)
+    - if @last_attempt
+      - last_attempt_entry = JSON.parse(@last_attempt).find{|item| item["level_id"]==level.id}
+      - last_attempt = last_attempt_entry["result"] if last_attempt_entry
+    - else
+      - last_attempt = ""
+    - level_class = level.class.to_s.underscore
 
-- if page < data["pages"].length - 1
-  %a.btn.btn-large.btn-primary.nextPageButton
-    =t('next_page')
-- else
-  = render partial: 'levels/dialog', locals: {app: app, data: data}
-  %a.btn.btn-large.btn-primary.unsubmitButton{style: "display: none"}
-    =t('unsubmit')
+    - level.properties['title'] = nil
 
+    - if level_class == "multi"
+      -# For students (or teachers doing PD), mark all answers incorrect, so we don't give anything away.
+      - unless current_user.try(:authorized_teacher?) && !@script.try(:pd?)
+        - level.properties['answers'].each { |answer| answer['correct'] = false }
+      = render partial: 'levels/single_multi', locals: {page: page, index: index, number: prior_page_count + index + 1, standalone: false, level: level, last_attempt: last_attempt, tight_layout: true }
```

Comment:
I felt same way, however, I rely on having index start at 0 for the JavaScript code that chugs through the results later on, and it seemed more complicated to change that.